File: 2021/10/27/main.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i, j, k = 0, 0, len(nums)-1
        while j <= k:
            n = nums[j]
            if n == 0:
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j += 1
            elif n == 1:
                j += 1
            elif n == 2:
                nums[j], nums[k] = nums[k], nums[j]
                k -= 1
        return nums

File: 2021/10/27/main.go
package main

func sortColors(nums []int) {
	i, j, k := 0, 0, len(nums)-1
	for j <= k {
		switch nums[j] {
		case 0:
			nums[i], nums[j] = nums[j], nums[i]
			i++
			j++
		case 1:
			j++
		case 2:
			nums[j], nums[k] = nums[k], nums[j]
			k--
		}
	}
}
